app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -325,10 +325,11 @@ def repository_types
 
     api :PUT, "/repositories/:id/republish", N_("Forces a republish of the specified repository, regenerating metadata and symlinks on the filesystem. Not allowed for repositories with the 'Complete Mirroring' mirroring policy.")
     param :id, :number, :desc => N_("Repository identifier"), :required => true
-    param :force, :bool, :desc => N_("Force metadata regeneration to proceed. (Deprecated)"), deprecated: true
+    param :force, :bool, :desc => N_("Force metadata regeneration to proceed. Dangerous when repositories use the 'Complete Mirroring' mirroring policy")
     def republish
-      if @repository.mirroring_policy == Katello::RootRepository::MIRRORING_POLICY_COMPLETE
-        fail HttpErrors::BadRequest, _("Metadata republishing is not allowed on repositories with the 'Complete Mirroring' mirroring policy.")
+      if @repository.mirroring_policy == Katello::RootRepository::MIRRORING_POLICY_COMPLETE && !::Foreman::Cast.to_bool(params[:force])
+        fail HttpErrors::BadRequest, _("Metadata republishing is dangerous on repositories with the 'Complete Mirroring' mirroring policy.
+          Change mirroring policy on the repository and try again or use 'force' parameter via hammer/API to stop mirroring upstream metadata and regenerate metadata locally.")
```

Comment:
This implies that using `force` will actually change the repo's mirroring policy. I know that's what happens in effect, but it doesn't _actually_ change the mirroring policy, right? Should it?

Reply:
It does not change the mirroring policy on repo object..I'll remove the `stop mirroring upstream metadata and` part of the message.

Reply:
Using force will regenerate metadata locally for now, but then continue mirroring upstream metadata on the next sync, right? Maybe we should mention that instead..

Reply:
`Metadata republishing is dangerous on repositories with the 'Complete Mirroring' mirroring policy. Change mirroring policy on the repository and try again if you want to stop using upstream metadata for this repository. Alternatively, you can use the 'force' parameter via hammer/API to regenerate metadata locally. This local metadata will be overwritten by upstream metadata on the next complete sync if repository mirroring policy is 'Complete Mirroring'.`